### DISCUSSION TITLE
BSD userland instead of GNU

### DISCUSSION THREAD

**Discussion by CRKatri** [ProcursusTeam/Procursus#730] [Ideas]
I have started some work on getting the while macOS/FreeBSD userland compiling for iOS at https://github.com/CRKatri/apple_cmds. What I have left to do is network_cmds (which is half done), system_cmds, and text_cmds. Once those are finished I will have to go back and fix all the BINDIRs so that they install to the correct location, then it should be ready, except for upgrade paths if this is something you’d be interested in for Pro. FYI although it uses BSD make it will only work with the version apple is maintaining, bmake will not work.

**Comment by CRKatri** [comment]
Besides that fact that not one of the binaries has been tested on a real devices, it is all done. All binaries (except pkill and a like one other) compile. All that’s left is testing it on a device then getting a transition from GNU to BSD working.

**Reply by CRKatri** [reply]
Oh, it also contains some binaries that are shipped with iOS, so it would definitely break your devices when not jailbroken.

**Reply by CRKatri** [reply]
I just tested it and it seems to work wonderfully, I also fixed any bin conflicts.

**Comment by Diatrus** [comment]
This is interesting. I would like, in the far future, to make BSD userland default and have GNU tools have the GNU prefix like on macOS.

**Reply by CRKatri** [reply]
"Far future"... So no time soon?
